Lib/fontmake/instantiator.py

```diff
@@ -224,21 +226,33 @@ def from_designspace(
         # because the math behind varLib and MutatorMath uses the default font as the
         # point of reference for all data.
         default_font = designspace.default.font
+        non_default_layer_name = designspace.default.layerName
+        non_default_layer_flag = (
+            non_default_layer_name is not None
+            and non_default_layer_name != DEFAULT_LAYER_NAME
```

Comment:
you should remove this `and non_default_layer_name != DEFAULT_LAYER_NAME`, it's not necessary. If the source.layerName is not None, then we can simply use it at face value, whatever that was. "foreground" or "public.default" aren't special (the latter is simply the name of an un-named default layer). What makes a layer the default is the fact that its directory is named 'glyphs' without any suffix.

Reply:
OK.